### ISSUE TITLE
Why not use the same scaling parameter 'alpha' for both train set and test set when stereo-rectified?

### ISSUE THREAD

**Issue by Tobyzai** [dimitrisPs/scared_toolkit#2]
Hello, thank you for sharing such a wonderful project!

In your repositories - [DeepPruner_SCARED](https://github.com/dimitrisPs/DeepPruner_SCARED):

'To recreate the SCARED fine-tuned model you should stereo-rectify the provided frames using alpha=0 as a parameter to the stereoRectify(). The same RGB stereo rectification process must be followed for the test set but with alpha=1 in order to preserve the whole rgb frame within the rectified view.'

I have a question, Why not use the same scaling parameter 'alpha' for both train set and test set when stereo-rectify?
Will there be any impact if the same parameters are used?

This puzzles me.

Looking forward to your reply.

**Comment by dimitrisPs** [comment]
Hello, 

You can definitely train with stereo-rectified frames using alpha=1 or any other alpha you wish. If I remember correctly, when I was training DeepPrunner in SCARED I was applying random crops to the input images and I didn't want to have big black borders resulting in black areas in my training samples. I never tested if this actually results in worse accuracy but my intuition was that training samples are already limited so having black patches in training, could potentially highlight the issue. Subsequent work followed the same workflow simply because I had a copy of the processed data ready and I knew this method works. 

On the other hand, If you do want to follow the evaluation protocol proposed in SCARED and the original depth frames, you should use an alpha that results in your input images being fully visible in your rectified view ( alpha=1 works in this case). This is because the evaluation protocol proposed in SCARED assigns a fixed penalty to pixels that do have reference information but no predictions. Using alpha=1 to rectify your images, after you estimate disparity for your stereo views, you can map disparity/depth back to the original unrectified/undistorted frames and still occupy the whole image area. Doing the same with frames rectified with alpha=0 for instance would result in black areas when you do the aforementioned inverse mapping. 

I hope this explains it.

**Comment by Tobyzai** [comment]
Got it， thank you very much!
